backend/app/modules/facades/current_user_budget.py

```diff
@@ -0,0 +1,24 @@
+from app.modules.octant_rewards import controller as octant_rewards_controller
+from app.modules.snapshots.pending import controller as pending_snapshot_controller
+from app.modules.user.budgets import controller as budgets_controller
+
+
+def get_current_user_budget(user_address: str) -> int:
+    """
+    This is a facade function that is used to get the current user budget.
+    Uses octant_rewards, pending_snapshot & budgets modules to get the current user budget.
+    """
+    estimated_eth_proceeds = (
```

Comment:
We don't have to create a facade for it and pass proceeds down the way through the flow. Please just construct a new service `EstimatedUserBudget` containing `get_budget` method, add it to `CURRENT` services and call it from the controller. Service should be constructed this way:
```
octant_rewards = CalculatedOctantRewards(
            staking_proceeds=EstimatedStakingProceeds(),
            effective_deposits=user_deposits,
        )
simulated_pending_snapshot_service = SimulatedPendingSnapshots(
            effective_deposits=user_deposits, octant_rewards=octant_rewards
        )
user_budgets = EstimatedUserBudgets(snapshots=simulated_pending_snapshot_service)
        ```

Reply:
good idea!